ruby-challenges.rb

```diff
@@ -3,25 +3,85 @@
 
 # --------------------1) Create a method that takes in an array of words and a single letter and returns an array of all the words containing that particular letter. Use the test variables provided.
 
-beverages_array = ['coffee', 'tea', 'juice', 'water', 'soda water']
+# Process
+# Create a method called "letter" that takes in an array and a letter as a parameter
+# create an empty array called "final_array" that will contain our desired values 
+# Use the array.each method to loop through each word in the provided arrays and determine if the words contain the letter
+# If the word contain the letter then the word will be added to our "final_array".
 
-letter_o = 'o'
-# Expected output: ['coffee', 'soda water']
-letter_t = 't'
+# beverages_array = ['coffee', 'tea', 'juice', 'water', 'soda water']
+
+# letter_o = 'o'
+# # Expected output: ['coffee', 'soda water']
+# letter_t = 't'
 # Expected output: ['tea', 'water', 'soda water']
 
+# def letter (array,letter)
+#     final_array = []
+#     array.each do |word|
+#         if word.include?(letter)
+#             final_array << word
+#         end
+#     end
+#     final_array
+# end
```

Comment:
I see you really like to create empty arrays / strings to push things into in order to create your final output. While this works just fine, I would challenge you to think of a way to not need that! For example, in this problem, we could use `.select` to only include words that include the letter, which would remove the need for an empty array to push words into. Just some food for thought, but nice work nonetheless!